pkg/cloudprovider/providers/gce/gce_backendservice.go

```diff
@@ -34,150 +38,113 @@ func newBackendServiceMetricContextWithVersion(request, region, version string)
 // GetGlobalBackendService retrieves a backend by name.
 func (gce *GCECloud) GetGlobalBackendService(name string) (*compute.BackendService, error) {
 	mc := newBackendServiceMetricContext("get", "")
-	v, err := gce.service.BackendServices.Get(gce.projectID, name).Do()
+	v, err := gce.c.BackendServices().Get(context.Background(), meta.GlobalKey(name))
 	return v, mc.Observe(err)
 }
 
 // GetAlphaGlobalBackendService retrieves alpha backend by name.
 func (gce *GCECloud) GetAlphaGlobalBackendService(name string) (*computealpha.BackendService, error) {
 	mc := newBackendServiceMetricContextWithVersion("get", "", computeAlphaVersion)
-	v, err := gce.serviceAlpha.BackendServices.Get(gce.projectID, name).Do()
+	v, err := gce.c.AlphaBackendServices().Get(context.Background(), meta.GlobalKey(name))
 	return v, mc.Observe(err)
 }
 
-// UpdateGlobalBackendService applies the given BackendService as an update to an existing service.
+// UpdateGlobalBackendService applies the given BackendService as an update to
+// an existing service.
 func (gce *GCECloud) UpdateGlobalBackendService(bg *compute.BackendService) error {
 	mc := newBackendServiceMetricContext("update", "")
-	op, err := gce.service.BackendServices.Update(gce.projectID, bg.Name, bg).Do()
-	if err != nil {
-		return mc.Observe(err)
-	}
-
-	return gce.waitForGlobalOp(op, mc)
+	return mc.Observe(gce.c.BackendServices().Update(context.Background(), meta.GlobalKey(bg.Name), bg))
 }
 
-// UpdateAlphaGlobalBackendService applies the given alpha BackendService as an update to an existing service.
+// UpdateAlphaGlobalBackendService applies the given alpha BackendService as an
+// update to an existing service.
 func (gce *GCECloud) UpdateAlphaGlobalBackendService(bg *computealpha.BackendService) error {
-	mc := newBackendServiceMetricContextWithVersion("update", "", computeAlphaVersion)
-	op, err := gce.serviceAlpha.BackendServices.Update(gce.projectID, bg.Name, bg).Do()
-	if err != nil {
-		return mc.Observe(err)
-	}
-
-	return gce.waitForGlobalOp(op, mc)
+	mc := newBackendServiceMetricContext("update", "")
+	return mc.Observe(gce.c.AlphaBackendServices().Update(context.Background(), meta.GlobalKey(bg.Name), bg))
 }
 
 // DeleteGlobalBackendService deletes the given BackendService by name.
 func (gce *GCECloud) DeleteGlobalBackendService(name string) error {
 	mc := newBackendServiceMetricContext("delete", "")
-	op, err := gce.service.BackendServices.Delete(gce.projectID, name).Do()
-	if err != nil {
-		if isHTTPErrorCode(err, http.StatusNotFound) {
-			return nil
-		}
-		return mc.Observe(err)
+	err := gce.c.BackendServices().Delete(context.Background(), meta.GlobalKey(name))
+	if isHTTPErrorCode(err, http.StatusNotFound) {
+		return nil
```

Comment:
Can cut. Both e2e/ingress-utils and ingress-gce check for NotFound.

Reply:
done